ops/testing.py

```diff
@@ -1804,8 +1865,28 @@ def action_log(self, message):  # type:ignore
     def action_fail(self, message=''):  # type:ignore
         raise NotImplementedError(self.action_fail)  # type:ignore
 
-    def network_get(self, endpoint_name, relation_id=None):  # type:ignore
-        raise NotImplementedError(self.network_get)  # type:ignore
+    def network_get(self, binding_name: str, relation_id: Optional[int] = None) -> '_NetworkDict':
+        if relation_id is None:
+            data = self._networks.get((binding_name, None))
+            if data is not None:
+                # Get the default data if it exists
+                return data
+            # Otherwise get the data with the lowest relation_id as the default
+            # TODO: does this even make sense?
+            datas = sorted((r, data) for (b, r), data in self._networks.items()
+                           if b == binding_name)
+            if datas:
+                return datas[0][1]
+        else:
+            data = self._networks.get((binding_name, relation_id))
+            if data is not None:
+                # Get the data for this specific relation_id if it exists
+                return data
+            data = self._networks.get((binding_name, None))
+            if data is not None:
```

Comment:
```suggestion
            if data:
```

Reply:
I actually think the explicit `is None` and `is not None` is better for values that can be None *or a value with another falsey value*. In this case `data` is a dict so could potentially be empty which would be falsey. (It shouldn't be in this specific case, but I think it's good practice.)